src/binary-reader-objdump.cc

```diff
@@ -903,6 +904,50 @@ Result BinaryReaderObjdumpDisassemble::OnOpcodeType(Type type) {
   return Result::Ok;
 }
 
+Result BinaryReaderObjdumpDisassemble::OnTryTableExpr(
+    Type sig_type,
+    const RawCatchVector& catches) {
+  if (!in_function_body) {
+    return Result::Ok;
+  }
+
+  std::string buffer = std::string();
+
+  if (sig_type != Type::Void) {
+    buffer.append(BlockSigToString(sig_type).c_str()).append(" ");
+  }
+
+  for (auto& catch_ : catches) {
+    switch (catch_.kind) {
+      case CatchKind::Catch:
+        buffer.append("catch ");
+        break;
+      case CatchKind::CatchRef:
+        buffer.append("catch_ref ");
+        break;
+      case CatchKind::CatchAll:
+        buffer.append("catch_all ");
+        break;
+      case CatchKind::CatchAllRef:
+        buffer.append("catch_all_ref ");
+        break;
+    }
+    if (catch_.kind == CatchKind::Catch || catch_.kind == CatchKind::CatchRef) {
+      buffer.append(std::to_string(catch_.tag));
+    }
+    buffer.append(" ").append(std::to_string(catch_.depth)).append(" ");
+  }
+
+  if (!buffer.empty()) {
+    // remove trailing space
+    buffer.pop_back();
+  }
```

Comment:
I wonder if there is some way to refactor this such to avoid adding and removing the trailing whitespace like this?